### PR TITLE
MDEV-50: add permission group viewer page

### PR DESCRIPTION
**Pull request by tcaiger**
### Changes:

- add a permission group viewer using the horizontal tree component

---

### Screenshots:
Screen Shot 2023-04-27 at 4 31 30 PM

### CODE REVIEW

**Review by acdunham — CHANGES_REQUESTED** [review]
Looks good! There are potentially son UX improvements we could make as part of this, or else just note down for a future ticket:
- If a permissions group list item doesn't have 'children' probably don't make that clickable, since nothing can happen with it unless it has children.
- Similarly, if a permissions group item has children, maybe making the whole list item clickable, instead of just the arrow might be nice, but if the above point was done then ithis would not be as important.


I also found a bug when filtering groups. To replicate:
1. Click a permission group in the first column
2. Filter the second column
3. Click a different group in the first column
4. Page goes blank, and error is:
image
(this may also happen at different stages of the tree, so might be worth checking when fixing)

File: packages/admin-panel/src/pages/resources/PermissionGroupsViewerPage.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    if (isLoading) {
+      return null;
+    }
+
+    if (node) {
+      return getDescendantData(node.id, data);
+    }
+
+    return getDescendantData(null, data);
+  }
```
[the remark this answers is not in the document]
Comment (tcaiger) [reply packages/admin-panel/src/pages/resources/PermissionGroupsViewerPage.js:55]: Nice suggestion thanks

**Comment by tcaiger** [comment]
> 

Good find on the bug, I hadn't encountered that yet, but I fixed it. In regards to the UX, I totally agree. The tree component was recently designed by Flic and developed by another dev so I'll chat to Flic and see if we can change it or if there is some reason for it being this way.

**Comment by acdunham** [comment]
> > 
> 
> Good find on the bug, I hadn't encountered that yet, but I fixed it. In regards to the UX, I totally agree. The tree component was recently designed by Flic and developed by another dev so I'll chat to Flic and see if we can change it or if there is some reason for it being this way.

Sounds good - I have approved anyway and if any UX changes come through as part of this you then can always re-review

**Comment by tcaiger** [comment]
@acdunham I talked to Flic and it turns out the tree is designed for a case where the selected items are editable which is why there is a difference in the interactions between the item text and the expand icon. So I just added a readOnly prop for the case I want and then updated the styling to match your suggestion. Would you mind reviewing the last commit I added?

**Review by acdunham — APPROVED** [review]

**Pull request merged**